### DISCUSSION TITLE
NPM Support?

### DISCUSSION THREAD

**Discussion by dopevelli** [buresdv/Cork#108] [Q&A]
Seeing in how you shouldn't use brew to install NPM because it is in it of itself a package manager, have you considered adding support for NPM as well?

_P.S._

_I really like it so far..._

**Comment by buresdv** [comment]
I'm glad you like it :)

Could you give me more information about what you mean? Do you want there to be a specific workflow for installing NPM that doesn't involve Homebrew at all?

I don't use NPM, so I don't know what the problems with it are 😅

**Comment by dopevelli** [comment]
So it's not recommended to install NPM, NVM or Node.js via Homebrew as NPM is a package manager in it of itself and NVM (Node Version Manager) is not supported by Homebrew. Instead it's recommended to install NVM which installs Node which installs NPM.

I was referring to adding NPM support in order to add, remove and update global and local NPM packages, search NPM registry and view NPM package info/registry status via Cork.

Here are some examples:

https://720kb.github.io/ndm/

https://www.npmjs.com/package/npm-gui

NVM support could be cool too (view, remove and update node versions).

Hope that clarifies things a bit...

**Comment by buresdv** [comment]
Hmm, that sounds very out of scope for what Cork is. 

I can imagine this maybe as a fork of Cork, but even then it would require the rewriting of basically the entire backend, which would turn it into a separate app altogether. I don't think I'm qualified enough to make this, since I don't use NPM or NVM at all.

But perhaps in the future if I ever start using it 😊
